src/Markdown/Editor/Impl/Classification/MdClassifier.cs

```diff
@@ -25,6 +26,8 @@ internal sealed class MdClassifier : TokenBasedClassifier<MarkdownTokenType, Mar
             ContentTypeRegistryService = contentTypeRegistryService;
             ClassificationNameProviders = classificationNameProviders;
             ClassificationRegistryService = classificationRegistryService;
+
+            ServiceManager.AddService<MdClassifier>(this, textBuffer);
```

Comment:
Usually it is only necessary if owning object is disposable OR the said service holds on native objects. Neither is the case here.